tests/prague/eip7702_set_code_tx/test_set_code_txs.py

```diff
@@ -0,0 +1,2555 @@
+        pytest.param(
+            1,
+            2**256 - 1,
+            1,
+            id="v=1,r=2**256-1,s=1",
+        ),
+    ],
+)
+def test_set_code_using_invalid_signatures(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    v: int,
+    r: int,
+    s: int,
+):
+    """
+    Test sending a transaction to set the code of an account using synthetic signatures.
+    """
+    success_slot = 1
+
+    callee_code = Op.SSTORE(success_slot, 1) + Op.STOP
+    callee_address = pre.deploy_contract(callee_code)
+
+    authorization_tuple = AuthorizationTuple(
+        address=0,
+        nonce=0,
+        chain_id=1,
+        v=v,
+        r=r,
+        s=s,
+    )
+
+    tx = Transaction(
+        gas_limit=100_000,
+        to=callee_address,
+        value=0,
+        authorization_list=[authorization_tuple],
+        error=TransactionException.TYPE_4_INVALID_AUTHORITY_SIGNATURE,
+        sender=pre.fund_eoa(),
+    )
+
+    state_test(
+        env=Environment(),
+        pre=pre,
+        tx=tx,
+        post={
+            callee_address: Account(
+                storage={success_slot: 0},
+            ),
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    "log_opcode",
+    [
+        Op.LOG0,
+        Op.LOG1,
+        Op.LOG2,
+        Op.LOG3,
+        Op.LOG4,
+    ],
+)
+@pytest.mark.with_all_evm_code_types
+def test_set_code_to_log(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    log_opcode: Op,
+):
+    """
+    Test setting the code of an account to a contract that performs the log operation.
+    """
+    sender = pre.fund_eoa()
+
+    set_to_code = (
+        Op.MSTORE(0, 0x1234)
+        + log_opcode(size=32, topic_1=1, topic_2=2, topic_3=3, topic_4=4)
+        + Op.STOP
+    )
+    set_to_address = pre.deploy_contract(set_to_code)
+
+    tx = Transaction(
+        gas_limit=10_000_000,
+        to=sender,
+        value=0,
+        authorization_list=[
+            AuthorizationTuple(
+                address=set_to_address,
+                nonce=1,
+                signer=sender,
+            ),
+        ],
+        sender=sender,
+    )
+
+    state_test(
+        env=Environment(),
+        pre=pre,
+        tx=tx,
+        post={
+            sender: Account(
+                nonce=2,
+                code=Spec.delegation_designation(set_to_address),
+            ),
+        },
+    )
+
+
+@pytest.mark.with_all_call_opcodes(
+    lambda opcode: opcode
+    not in [Op.STATICCALL, Op.CALLCODE, Op.DELEGATECALL, Op.EXTDELEGATECALL, Op.EXTSTATICCALL]
+)
+@pytest.mark.with_all_precompiles
+def test_set_code_to_precompile(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    precompile: int,
+    call_opcode: Op,
+):
+    """
+    Test setting the code of an account to a pre-compile address.
+    """
+    auth_signer = pre.fund_eoa(auth_account_start_balance)
+
+    caller_code_storage = Storage()
+    caller_code = (
+        Op.SSTORE(caller_code_storage.store_next(True), call_opcode(address=auth_signer))
```

Comment:
Some precompiles fail, if not provided valid input, and this expects the call to always succeed, how is it supposed to work? In particular, point evaluation precompile, blake2f, ec*.

Reply:
I see, I assumed it executes the precompile, EIP doesn't mention this case explicitly.

Reply:
@gumb0 I assumed this as well, and EthJS likely does this. The test, however, does not call into the precompile.


```suggestion
        Op.SSTORE(caller_code_storage.store_next(True), call_opcode(address=Address(precompile)))
```

Reply:
Never mind, this is actually correct :sweat_smile:

Reply:
Could we add a storage check to check if the call succeeds? It currently checks for empty storage, but I think it should explicitly check for slot 0 (call result) = 1 and slot 1 (RETURNDATASIZE) = 0.